hieroplatform/src/main/java/org/hiero/remoting/HieroServer.java

```diff
@@ -0,0 +1,204 @@
+package org.hiero.remoting;
+
+import com.google.common.net.HostAndPort;
+import com.google.protobuf.ByteString;
+import io.grpc.Server;
+import io.grpc.netty.NettyServerBuilder;
+import io.grpc.stub.StreamObserver;
+import org.apache.commons.lang3.SerializationUtils;
+import org.hiero.dataset.api.IDataSet;
+import org.hiero.dataset.api.PartialResult;
+import org.hiero.pb.Ack;
+import org.hiero.pb.Command;
+import org.hiero.pb.HieroServerGrpc;
+import org.hiero.pb.PartialResponse;
+import org.hiero.utils.Converters;
+import rx.Observable;
+import rx.Subscriber;
+import rx.Subscription;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Server that transfers map(), sketch(), zip() and unsubscribe() RPCs from a RemoteDataSet
+ * object to locally managed IDataSet objects, and streams back results.
+ */
+public class HieroServer extends HieroServerGrpc.HieroServerImplBase {
+    public static final int DEFAULT_IDS_INDEX = 1;
+    private static final int NUM_THREADS = 5;
+    private static final Executor EXECUTOR = Executors.newFixedThreadPool(NUM_THREADS);
+    private final Server server;
+    private final AtomicInteger dsIndex = new AtomicInteger(0);
+    private final ConcurrentHashMap<Integer, IDataSet> dataSets;
+    private final ConcurrentHashMap<UUID, Subscription> operationToObservable
+            = new ConcurrentHashMap<>();
+    private final HostAndPort listenAddress;
+
+    public HieroServer(final HostAndPort listenAddress, final IDataSet dataSet) throws IOException {
+        this.listenAddress = listenAddress;
+        this.server = NettyServerBuilder.forAddress(new InetSocketAddress(listenAddress.getHost(),
+                                                                     listenAddress.getPort()))
+                                        .executor(EXECUTOR)
+                                        .addService(this)
+                                        .build()
+                                        .start();
+        this.dataSets = new ConcurrentHashMap<>();
+        this.dataSets.put(this.dsIndex.incrementAndGet(), dataSet);
+    }
+
+    /**
+     * Implementation of map() service in hiero.proto.
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public void map(final Command command, final StreamObserver<PartialResponse> responseObserver) {
+        if (!checkValidIdsIndex(command.getIdsIndex(), responseObserver)) {
+            return;
+        }
+        final byte[] bytes = command.getSerializedOp().toByteArray();
+        final MapOperation mapOp = SerializationUtils.deserialize(bytes);
```

Comment:
What happens if any of these fail?
Maybe you want a try-catch so you can call onError for the responseObserver